syntax = "proto3";

service AuthService {
  rpc AuthenticateUser (UserCredentials) returns (AuthToken);
  rpc ValidateToken (Token) returns (TokenInfo);
  rpc InvalidateToken (Token) returns (SuccessResponse);
}

message UserCredentials {
  string username = 1;
  string password = 2;
}

message AuthToken {
  string token_value = 1;
}

message Token {
  string token_value = 1;
}

message TokenInfo {
  string sub = 1;
  repeated string role = 2;
}

message SuccessResponse {
  string message = 1;
}
